specs/interop/fault_proof.md

```diff
@@ -9,16 +9,292 @@
+but now extended with a `messages_root`.
+The `messages_root` commits to accumulators of the initiating messages and executing messages.
+
+```python
+payload = state_root || withdrawal_storage_root || latest_block_hash || messages_root
+version_byte = bytes32(uint256(1))  # Big-endian. Name is spec legacy, it is 32 bytes long.
+output_root = keccak256(version_byte || payload)
+```
+
+## Bisection
+
+No changes to the dispute game bisection are required.
+The only changes required are to the [fault proof program](#program) itself.
+The program interprets the bisection in a new way,
+following the [Decision Tree](#decision-tree) to identify the sub-problem to prove.
+
+### Entering the VM bisection
+
+The bisection-game specifies a fixed-depth where the form of dispute changes from 
+application-layer constructs to VM emulation.
+
+Depth is defined such that `depth = 0` is the root of the tree, and each nested counter-claim increments the depth by 1.
+
+Between depth 0 and the `SPLIT_DEPTH` (incl.) the claims represent chain constructs, such as output-roots.
+
+From `SPLIT_DEPTH` to `MAX_DEPTH` the bisection claims represent pivots that commit to the VM-state
+at the half-way point in the corresponding VM-execution-trace scope.
+
+### Pivots in bisection
+
+When bisecting the MIPS VM trace of Cannon, the concept of "Pivots" is used.
+
+Pivots allow both sides of the game to argue about state snapshots in the trace,
+without having to fully compute commitments for all intermediate states before or after the snapshots.
+
+When a pivot is agreed upon, the intermediate steps that both sides took are irrelevant,
+and a common pre-state is thus established quickly.
+
+## Decision Tree
+
+To facilitate interop between many chains, we interpret the bisection path to agree on a sub-problem to solve.
+The decision-tree defines how the bisection path is interpreted, and thus how the sub-problem is identified.
+
+The in-flight bisection identifies claims in two ways:
+- relative their parent-claim, to facilitate a multi-player dispute game
+- absolute to the root, to statically point to what is being disputed in the reference frame of the player.
+
+The absolute identifier is typed as a [`Position`](../experimental/fault-proof/fault-dispute-game.md#position),
+representing a generalized index, a unique number identifying the position in the binary tree.
+
+From this position, we can derive a `depth`, and a `path`:
+- The `depth` is the bisection-round relative to the root of the tree.
+  This is the bit-length of the generalized index.
+- The `path` is a bitfield, of left/right choices, one per `depth`,
+  starting from the root of the bisection game, up to the `depth`.
+  This the remaining bitstring after the leading `1` bit of the generalized index.
+
+TODO: diagram of the binary tree
+
+`SPLIT_DEPTH = 64`
+
+- `path[0]`: [L1 Execution](#l1-execution-extra) [Experimental]
+- `path[0] == 0`: L1 block-number bisection
```

Comment:
I think it would be helpful with typing if we used `path[0] == 0b0`, ie the `0b` prefix where it makes sense, it helps me keep in context what things are